raster-foundry/raster-foundry
# Add GeoTiff info on COG post

## Overview

This PR makes it so that we spin off fetching GeoTiff info for new COGs that have been posted into a separate thread that doesn't block the server's response. It picks up the `Blocker` added in #5465 and uses it to get the GeoTiffInfo and moves the GDALRasterSource calls out of that blocker and back into the default execution context (but still wrapped in `F`).

### Checklist


## Testing Instructions

- Here's a nice json for you -- save it in `scene.json`

```json
{
	"id": null,
	"visibility": "PRIVATE",
	"tags": [],
	"datasource": "e4d1b0a0-99ee-493d-8548-53df8e20d2aa",
	"sceneMetadata": {},
	"name": "Test scene",
	"owner": null,
	"tileFootprint": null,
	"dataFootprint": null,
	"metadataFiles": [],
	"images": [],
	"thumbnails": [],
	"ingestLocation": "s3://rasterfoundry-development-data-us-east-1/cog-pa_m_4208063_sw_17_1_20150727.tif",
	"filterFields": {},
	"statusFields": {
		"ingestStatus": "NOTINGESTED",
		"thumbnailStatus": "SUCCESS",
		"boundaryStatus": "SUCCESS"
	},
	"sceneType": "COG"
}
```
- assemble api server and tile server
- set the `api-server` service's log level to DEBUG in docker-compose
- `./scripts/server`
- Copy the "RF Dev user refresh token" from Lastpass
- `echo {"refresh_token": "the refresh token"} | http :9100/api/tokens`
- `export JWT_AUTH_TOKEN=<the id_token that you got back>`
- `cat scene.json | http --auth-type=jwt :9100/api/scenes`
- note that in your response the scene is not ingested
- but also note in the logs that within not very long, you see a nice log message from the `geotiffinfo` thread pool letting you know that it got info for that URI
- now request the scene by id -- `http --auth-type=jwt :9100/api/scenes/<your scene id> | jq .statusFields`
- it should now show as INGESTED

Closes #5466 

Closes #5465
